database/tbcd/level/headercache.go

```diff
@@ -47,9 +50,40 @@ func (l *lowIQMap) Get(k *chainhash.Hash) (*tbcd.BlockHeader, bool) {
 	return bh, ok
 }
 
-func lowIQMapNew(count int) *lowIQMap {
+func (l *lowIQMap) Purge(k *chainhash.Hash) {
+	l.mtx.Lock()
+	defer l.mtx.Unlock()
+
+	delete(l.m, *k)
+}
+
+func (l *lowIQMap) PurgeBatch(ks []*chainhash.Hash) {
+	l.mtx.Lock()
+	defer l.mtx.Unlock()
+
+	for v := range ks {
+		delete(l.m, *ks[v])
```

Comment:
Although I don't think that it's possible at this point for this pointer to be `nil`, I am going to ask that we make sure that this pointer is not `nil` before de-referencing it (perhaps log a warning as well).

Also, does the input need to be a slice of pointers? or could we just do `[]chainhash.Hash`?

Reply:
We need to punish the caller if the pointer is nil. As it is written now it cannot be nil. Also, these operations are already expensive so not blowing the pre-fetcher is also nice :-)